javascript/script.js

```diff
@@ -0,0 +1,242 @@
+		$("list-items").insertAdjacentHTML('afterbegin', emptyImage);
+		$("list-items").classList.add("add-empty-image");
+	}
+	if (listReceived) {
+		for (let i = 0; i < listReceived.length; i++) {
+			$("search-text").classList.remove("hidden");
+			addToDoItem(listReceived[i]);
+		}
+	}
+
+	//error message on closing tab while editing
+	window.onbeforeunload = () => {
+		let confirmationMessage = 'Are you sure you want to leave?';
+		if ((newItemInput.value.trim() !== "") || toDoList.some((el) => el.editMode)) {
+			return confirmationMessage;
+		}
+	}
+	$("clear-search").classList.add("hidden");
+});//end of event on load
+
+//checkbox functionalities and marking it as complete in list
+function toggleCheckedClass(targetId) {
+	let checkBox = $("check-" + targetId)
+	let description = $("desc-" + targetId)
+	if (checkBox.checked)
+		description.classList.add("strike-through");
+	else
+		description.classList.remove("strike-through");
+	toDoList.find(element => element.id === targetId).taskCompleted = checkBox.checked;
+	saveItemsToLocalStorage();
+}
+
+let searchInput = $("search-text");
+$("clear-search").addEventListener("click", () => {
+	searchInput.value = ""
+	let event = new Event('input', {
+		bubbles: false,
+		cancelable: true,
+	});
+	searchInput.dispatchEvent(event);
+	$("clear-search").classList.add("hidden");
+});
+
+let count;
+searchInput.addEventListener("input", e => {
+	count = 0;
+	let textToBeSearched = e.target.value.trim().toLowerCase();
+	$("clear-search").classList.toggle("hidden", !textToBeSearched);
+
+	toDoList.forEach(item => {
+		const isVisible = item.description.toLowerCase().includes(textToBeSearched);
+		$('item-' + item.id).classList.toggle("hidden", !isVisible);
+		if (isVisible) {
+			count++;
+		}
+	})
+
+	let noResultsFoundImage = $("noResultsFoundImage");
+	if (count === 0 && !noResultsFoundImage) {
+		let noResultsFoundImageHtml = "<figure id='noResultsFoundImage'><img class='empty' src='./images/emptySearch.svg'><figcaption>No results found...</figcaption></figure>";
+		$("list-items").insertAdjacentHTML('afterbegin', noResultsFoundImageHtml);
+		$("list-items").classList.add("add-empty-image");
+	}
+	else if (count > 0 && noResultsFoundImage) {
+		noResultsFoundImage.remove();
+		$("list-items").classList.remove("add-empty-image");
+	}
+});
+function displayInputBar() {
+	$("float-div").classList.add("float-div");
+	$("float-button").classList.add("hidden");
+	$("input-area").classList.add("flex");
+	newItemInput.focus();
+}
+function displayFab() {
+	$("overlay").classList.remove("flex");
+	$("float-div").classList.remove("float-div");
+	$("float-button").classList.remove("hidden");
+	$("input-area").classList.remove("flex");
+}
+$("plus").addEventListener("click", displayInputBar)
+$("float-button").addEventListener("click", displayInputBar)
+$("float-div").addEventListener("click", displayFab);
```

Comment:
Do not do DOM operations in the global scope. We need to make sure that the DOM is loaded before trying to access it. So move these into the `onload` event handler.